src/mmw/apps/bigcz/clients/cuahsi.py

```diff
@@ -45,16 +77,43 @@ def find_service(services, service_code):
     return None
 
 
-def parse_records(sites, services):
+def parse_records(series, services):
+    """
+    Join series catalog records to corresponding service.
+    """
     result = []
-    for site in sites:
-        service = find_service(services, site['servCode'])
+    for record in series:
+        service = find_service(services, record['ServCode'])
         if service:
-            record = parse_record(site, service)
+            record = parse_record(record, service)
             result.append(record)
     return result
 
 
+def group_series_by_location(series):
```

Comment:
Because there are no aggregated fields, it's fair to say that this is more of a distinct operation rather than a grouping. They are very similar operations. The comparable SQL query may look like:

```sql
SELECT ServCode, ServURL, location, ...
FROM series
GROUP BY ServCode, ServURL, location, ...
```